### PR TITLE
nmcli: new arguments to ignore automatic dns servers and gateways

### LINKED ISSUE #1087: nmcli: new arguments to ignore automatic dns servers and gateways
##### SUMMARY

It'd be nice to be able to control whether routes and dns servers from automatic sources are ignored.  This is especially handy if you have multiple DHCP connections and wish to ignore the DNS servers and gateways from the others.  Basically to add the following new arguments:

| `nmcli` module argument | NetworkManager setting |
|-------------------------|------------------------| 
| `dns4_ignore_auto` | `ipv4.ignore-auto-dns` |
| `gw4_ignore_auto` | `ipv4.ignore-auto-routes` |
| `dns6_ignore_auto` | `ipv6.ignore-auto-dns` |
| `gw6_ignore_auto` | `ipv6.ignore-auto-routes` |

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

`nmcli`

##### ADDITIONAL INFORMATION

```yaml
- name: 'primary connection'
  nmcli:
    conn_name: primary
    type: ethernet
    ifname: eth0
- name: 'secondary connection'
  nmcli:
    conn_name: secondary
    type: ethernet
    ifname: eth1
    dns4_ignore_auto: true
    dns6_ignore_auto: true
    gw4_ignore_auto: true
    gw6_ignore_auto: true
```

Note: these would be nice to have in implementing this feature:
* The IPv4 and IPv6 methods could be explicitly set (e.g., to `auto`).
* The `Nmcli` connection creation/modification methods modified IP settings consistently across connection types.

### PR DESCRIPTION
**Pull request by yan12125**
##### SUMMARY

nmcli: new arguments to ignore automatic dns servers and gateways

Closes #1087

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
nmcli

##### ADDITIONAL INFORMATION

Manually tested. With the following task:
```YAML
- name: configure ethernet connection
  community.general.nmcli:
    conn_name: Ethernet
    state: present
    method4: auto
    method6: auto
    dns4_ignore_auto: true
    dns6_ignore_auto: true
```

I got the following output from nmcli:

```
$ LANG=en_US.UTF-8 nmcli connection show Ethernet | grep ignore
ipv4.ignore-auto-routes:                no
ipv4.ignore-auto-dns:                   yes
ipv6.ignore-auto-routes:                no
ipv6.ignore-auto-dns:                   yes
```

And in nm-connection-editor, I got "Automatic (DHCP) addresses only" and "Automatic, addresses only" in "IPv4 Settings" and "IPv6 Settings", respectively.

### CODE REVIEW

File: changelogs/fragments/2635-nmcli-add-ignore-auto-arguments.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Add new arguments to ignore automatic dns servers and gateways in nmcli (https://github.com/ansible-collections/community.general/issues/1087).
```
Comment (felixfontein) [inline changelogs/fragments/2635-nmcli-add-ignore-auto-arguments.yml:2]:
```suggestion
  - nmcli - add new options to ignore automatic DNS servers and gateways (https://github.com/ansible-collections/community.general/issues/1087).
```

File: plugins/modules/net_tools/nmcli.py [hunk 2]
```diff
@@ -77,6 +77,12 @@
             - Use the format C(192.0.2.1).
             - This parameter is mutually_exclusive with never_default4 parameter.
         type: str
+    gw4_ignore_auto:
+        description:
+            - Ignore automatically configured IPv4 routes.
+        type: bool
+        default: no
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:84]:
```suggestion
        default: false
```
Please use `true`/`false` for YAML booleans.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by yan12125** [comment]
Done! Thanks for suggestions!

**Review by aminvakil — APPROVED** [review]
Thanks for implementing this!
I don't use this module myself, but change LGTM.

**Comment by felixfontein** [comment]
@yan12125 thanks for implementing this!
@aminvakil thanks for reviewing!

**Pull request merged**